library/src/main/java/com/novoda/downloadmanager/DownloadBatch.java

```diff
@@ -260,4 +261,16 @@ void persist() {
                 downloadBatchStatus.notificationSeen()
         );
     }
+
+    @WorkerThread
+    void persist() {
```

Comment:
I thought this one wasn't necessary? Only called from one place and that should be async too?